android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java

```diff
@@ -148,10 +148,16 @@ public void onNotified(RPCNotification notification) {
 				if (onHMIStatus.getWindowID() != null && onHMIStatus.getWindowID() != PredefinedWindows.DEFAULT_WINDOW.getValue()) {
 					return;
 				}
+				HMILevel prevHMILevel = hmiLevel;
 				hmiLevel = onHMIStatus.getHmiLevel();
 				if(hmiLevel.equals(HMILevel.HMI_FULL)){
 					checkState();
 				}
+				if (hasStarted && (prevHMILevel.equals(HMILevel.HMI_FULL) || prevHMILevel.equals(HMILevel.HMI_LIMITED)) && (hmiLevel.equals(HMILevel.HMI_NONE) || hmiLevel.equals(HMILevel.HMI_BACKGROUND))){
```

Comment:
Potential NPE using the `equals` method here. You can either use the `==` comparator  or switch the enum instance calling the `equals()` metod, eg  `HMILevel.HMI_FULL.equals(prevHMILevel)` as the `HMILevel` enum instances will always be valid. Either solution will work.